src/openms/include/OpenMS/SYSTEM/BuildInfo.h

```diff
@@ -0,0 +1,225 @@
+      String getOSAsString() const
+      {
+        return OpenMS_OSNames[os_];
+      }
+
+      /// @brief Get the current architecture (32-bit or 64-bit)
+      String getArchAsString() const
+      {
+        return OpenMS_ArchNames[arch_];
+      }
+
+      /// @brief Get the OS version (e.g. 10.15 for macOS or 10 for Windows)
+      String getOSVersionAsString() const
+      {
+        return os_version_;
+      }
+
+      /// @brief Get Architecture of this binary (simply by looking at size of a pointer, i.e. size_t).
+      static String getBinaryArchitecture()
+      {
+        size_t bytes = sizeof(size_t);
+        switch (bytes)
+        {
+          case 4:
+            return OpenMS_ArchNames[ARCH_32BIT];
+          case 8:
+            return OpenMS_ArchNames[ARCH_64BIT];
+          default:
+            return OpenMS_ArchNames[ARCH_UNKNOWN];
+        }
+      }
+
+      /// @brief Constructs and returns an OpenMSOSInfo object
+      static OpenMSOSInfo getOSInfo()
+      {
+        OpenMSOSInfo info;
+        #if defined(WIN32)  // Windows
+        info.os_ = OS_WINDOWS;
+        info.arch_ = getArchOnWin();
+        info.os_version_ = getWinOSVersion();
+        #elif (defined(__MACH__) && defined(__APPLE__)) // MacOS
+        info.os_ = OS_MACOS;
+        #else //Linux
+        info.os_ = OS_LINUX;
+        #endif
+        // check if we can use QSysInfo
+        #if (defined(Q_OS_MACOS) || defined(Q_OS_UNIX))
+        info.os_version_ = QSysInfo::productVersion();
+        // identify architecture
+        if (QSysInfo::WordSize == 32)
+        {
+          info.arch_ = ARCH_32BIT;
+        }
+        else
+        {
+          info.arch_ = ARCH_64BIT;
+        }
+        #endif
+        return info;
+      }
+
+      //********************
+      //  Windows specific API calls
+      //********************
+      #ifdef WIN32
+      #include <windows.h>
+      #include <stdio.h>
+
+      typedef BOOL (WINAPI * LPFN_ISWOW64PROCESS)(HANDLE, PBOOL);
+
+      LPFN_ISWOW64PROCESS fnIsWow64Process;
+
+      OpenMS_Architecture getArchOnWin()
+      {
+        #ifdef OPENMS_64BIT_ARCHITECTURE
+        return ARCH_64BIT;
+
+        #else
+        BOOL bIsWow64 = FALSE;
+
+        //IsWow64Process is not available on all supported versions of Windows.
+        //Use GetModuleHandle to get a handle to the DLL that contains the function
+        //and GetProcAddress to get a pointer to the function if available.
+
+        fnIsWow64Process = (LPFN_ISWOW64PROCESS) GetProcAddress(
+          GetModuleHandle(TEXT("kernel32")), "IsWow64Process");
+
+        if (NULL != fnIsWow64Process)
+        {
+          if (!fnIsWow64Process(GetCurrentProcess(), &bIsWow64))
+          {
+            return ARCH_UNKNOWN;
+          }
+        }
+        if (bIsWow64)
+        {
+          return ARCH_64BIT;
+        }
+        else
+        {
+          return ARCH_32BIT;
+        }
+        #endif
+      }
+
+      String getWinOSVersion()
```

Comment:
maybe comment why `QSysInfo::productVersion()` is not good enough? (lacking minor/path information?)

Reply:
Ah, actually I don't know why? :D Maybe it was just not present when this function was written?

Reply:
TBH personally I am not sure if I ever would need more information than https://doc.qt.io/qt-5/qsysinfo.html#productVersion

Reply:
Let's go with just the Qt Info.